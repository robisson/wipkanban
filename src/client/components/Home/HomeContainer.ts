import Home from "./index";
import { connect } from "react-redux";

const mapStateToProps = ({ userReducer: { user } }:any) => ({ user });

const HomeContainer = connect(mapStateToProps)(Home);

export default HomeContainer;
