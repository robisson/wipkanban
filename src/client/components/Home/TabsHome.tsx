import React from "react";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import PropTypes from "prop-types";
import Typography from "@material-ui/core/Typography";
import KanbanHome from "../KanbanHome";

interface ITabContainerProps {
  children: any;
}

function TabContainer(props: ITabContainerProps) {
  return (
    <Typography component="div" style={{ padding: 8 * 3 }}>
      {props.children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired
};

interface ITabsHomeState {
  value: number;
}

class TabsHome extends React.Component<{}, ITabsHomeState> {
  state = {
    value: 0
  };

  handleChange = (event: any, value: number) => {
    this.setState({ value });
  };

  render() {
    const { value } = this.state;
    return (
      <div style={{ width: "100%" }}>
        <Paper>
          <Tabs
            value={value}
            onChange={this.handleChange}
            indicatorColor="secondary"
            textColor="primary"
          >
            <Tab label="Kanban boards" />
            <Tab label="OKRs" />
          </Tabs>
        </Paper>
        <div>
          {value === 0 && (
            <TabContainer>
              <KanbanHome />
            </TabContainer>
          )}
          {value === 1 && <TabContainer>ITEM 2</TabContainer>}
        </div>
      </div>
    );
  }
}

export default TabsHome;
