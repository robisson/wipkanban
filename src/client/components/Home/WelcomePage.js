// @flow
import React from "react";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import CssBaseline from "@material-ui/core/CssBaseline";
import Paper from "@material-ui/core/Paper";
import GridContainer from "../GridContainer";
import Button from "@material-ui/core/Button";

const styles: Object = theme => ({
  layout: {
    width: "auto",
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit * 2,
    [theme.breakpoints.up(800 + theme.spacing.unit * 2 * 2)]: {
      width: "50%",
      marginLeft: "auto",
      marginRight: "auto"
    }
  },
  h1: {
    width: "100%"
  },
  paper: {
    marginBottom: theme.spacing.unit * 3,
    padding: theme.spacing.unit * 2,
    [theme.breakpoints.up(600 + theme.spacing.unit * 3 * 2)]: {
      marginBottom: theme.spacing.unit * 6,
      padding: theme.spacing.unit * 3
    }
  },
  buttons: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-end"
  },
  button: {
    marginLeft: theme.spacing.unit
  }
});

type Props = {
  classes: Object,
  onLogout: Function,
  onUpdateAccountUser: Function,
  user: Object
};

const WelcomePage = ({
  classes,
  onLogout,
  onUpdateAccountUser,
  user
}: Props) => {
  user.firstAccess = false;

  return (
    <React.Fragment>
      <CssBaseline />
      <GridContainer>
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <Typography
              className={classes.h1}
              component="h1"
              variant="h2"
              align="center"
              color="textPrimary"
              gutterBottom
            >
              Welcome to WipLean
            </Typography>
            <Typography variant="subtitle1" align="center">
              Before you start using WipLean, please complete your data below.
            </Typography>
            <Grid container spacing={24}>
              <Grid item lg={6}>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.name = target.value)}
                    defaultValue={user.name}
                    required
                    label="First name"
                    fullWidth
                    autoComplete="fname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.lastname = target.value)}
                    defaultValue={user.lastname}
                    required
                    label="Last name"
                    fullWidth
                    autoComplete="lname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.email = target.value)}
                    defaultValue={user.email}
                    required
                    label="Email"
                    fullWidth
                    autoComplete="billing address-level2"
                  />
                </Grid>
              </Grid>
              <Grid item lg={6}>
                <div
                  style={{
                    backgroundColor: "red",
                    width: "100%",
                    height: 250
                  }}
                />
                <input
                  type="file"
                  onChange={({ target }) => (user.image = target.files[0])}
                />
              </Grid>
              <Grid item className={classes.buttons}>
                <Button onClick={onLogout}>Logout</Button>
                <Button
                  onClick={() => onUpdateAccountUser(user)}
                  variant="contained"
                  color="primary"
                  className={classes.button}
                >
                  Update and next
                </Button>
              </Grid>
            </Grid>
          </Paper>
        </main>
      </GridContainer>
    </React.Fragment>
  );
};

export default withStyles(styles)(WelcomePage);
