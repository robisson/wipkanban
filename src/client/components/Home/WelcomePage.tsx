import * as React from "react";
import Typography from "@material-ui/core/Typography";
import {
  withStyles,
  StyleRulesCallback,
  Theme
} from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import CssBaseline from "@material-ui/core/CssBaseline";
import Paper from "@material-ui/core/Paper";
import GridContainer from "../GridContainer";
import Button from "@material-ui/core/Button";
import grey from "@material-ui/core/colors/grey";

const styles: StyleRulesCallback = (theme: Theme) => ({
  layout: {
    width: "auto",
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit * 2,
    [theme.breakpoints.up(800 + theme.spacing.unit * 2 * 2)]: {
      width: "50%",
      marginLeft: "auto",
      marginRight: "auto"
    }
  },
  h1: {
    width: "100%"
  },
  paper: {
    marginBottom: theme.spacing.unit * 3,
    padding: theme.spacing.unit * 2,
    [theme.breakpoints.up(600 + theme.spacing.unit * 3 * 2)]: {
      marginBottom: theme.spacing.unit * 6,
      padding: theme.spacing.unit * 3
    }
  },
  buttons: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-end"
  },
  button: {
    marginLeft: theme.spacing.unit
  },
  photo: {
    backgroundColor: grey[300],
    width: 150,
    height: 200,
    margin: [`${theme.spacing.unit * 2}px`, "auto"].join(" ")
  }
});

interface IWelcomePageProps {
  classes: any;
  onLogout: any;
  onUpdateAccountUser: Function;
  user: any;
}

const WelcomePage: React.StatelessComponent<any> = ({
  classes,
  onLogout,
  onUpdateAccountUser,
  user
}: IWelcomePageProps) => {
  user.firstAccess = false;

  return (
    <React.Fragment>
      <CssBaseline />
      <GridContainer>
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <Typography
              className={classes.h1}
              component="h1"
              variant="h2"
              align="center"
              color="textPrimary"
              gutterBottom
            >
              Welcome to WipLean
            </Typography>
            <Typography variant="subtitle1" align="center">
              Before you start using WipLean, please complete your data below.
            </Typography>
            <Grid container spacing={24}>
              <Grid item lg={6}>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.name = target.value)}
                    defaultValue={user.name}
                    required
                    label="First name"
                    fullWidth
                    autoComplete="fname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.lastname = target.value)}
                    defaultValue={user.lastname}
                    required
                    label="Last name"
                    fullWidth
                    autoComplete="lname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.email = target.value)}
                    defaultValue={user.email}
                    required
                    label="Email"
                    fullWidth
                    autoComplete="billing address-level2"
                  />
                </Grid>
              </Grid>
              <Grid item lg={6}>
                <div className={classes.photo} id="divGreyPhoto">
                  <img id="imagePreview" width="100%" />
                </div>
                <input
                  type="file"
                  id="contained-button-file"
                  onChange={({ target }: any) => {
                    user.image = target.files[0];

                    let reader = new FileReader();

                    reader.onload = function(e: any) {
                      // get loaded data and render thumbnail.
                      const image: HTMLElement = document.getElementById(
                        "imagePreview"
                      ) as HTMLElement;

                      if (!!image) {
                        (image as HTMLImageElement).src = e.target.result;
                      }
                    };

                    // read the image file as a data URL.
                    reader.readAsDataURL(target.files[0]);
                  }}
                  accept="image/*"
                  style={{ display: "none" }}
                />
                <div style={{ textAlign: "center" }}>
                  <label htmlFor="contained-button-file">
                    <Button variant="contained" component="span">
                      Choose image
                    </Button>
                  </label>
                </div>
              </Grid>
              <Grid item className={classes.buttons}>
                <Button onClick={onLogout}>Logout</Button>
                <Button
                  onClick={() => onUpdateAccountUser(user)}
                  variant="contained"
                  color="primary"
                  className={classes.button}
                >
                  Update and next
                </Button>
              </Grid>
            </Grid>
          </Paper>
        </main>
      </GridContainer>
    </React.Fragment>
  );
};

export default withStyles(styles)(WelcomePage);
