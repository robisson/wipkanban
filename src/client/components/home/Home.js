import React from 'react';
import NavbarContainer from '../navbar/NavbarContainer';
import {Link} from 'react-router-dom';
import SubNav from './SubNav';

const Home = () => {
    return (
        <div>
            <NavbarContainer/>
            <SubNav/><br/>
            <div className="container-fluid">
                <div className="row">
                    <div className="col">
                        <h6>
                            My companies&nbsp;|&nbsp;
                            <Link to="/companys" className="text-murrey">View all</Link>
                        </h6>
                        <h5>You do not have any registered companies yet.</h5><br/>
                    </div>
                </div>
                <div className="row">
                    <div className="col">
                        <h6>
                            My projetcs&nbsp;|&nbsp;
                            <Link to="/projetcs" className="text-murrey">View all</Link>
                        </h6>
                        <h5>You do not have any registered projects yet.</h5><br/>
                    </div>
                </div>
                <div className="row">
                    <div className="col">
                        <h6>
                            My boards&nbsp;|&nbsp;
                            <Link to="/boards" className="text-murrey">View all</Link>
                        </h6>
                        <h5>You do not have any registered boards yet.</h5><br/>
                    </div>
                </div>
                <div className="row">
                    <div className="col">
                        <h6>
                            My teams&nbsp;|&nbsp;
                            <Link to="/boards" className="text-murrey">View all</Link>
                        </h6>
                        <h5>You do not have any registered teams yet.</h5><br/>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default Home;