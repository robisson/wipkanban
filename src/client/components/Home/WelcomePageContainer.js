import WelcomePage from "./WelcomePage";
import { connect } from "react-redux";
import { logout } from "../../actions/Login";

const mapStateToProps = ({ userReducer: { user } }) => ({ user });

const mapDispatchToProps = dispatch => {
  return {
    onLogout() {
      dispatch(logout());
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WelcomePage);
