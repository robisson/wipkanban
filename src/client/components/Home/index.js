// @flow
import React from "react";
import Navbar from "../Navbar";
import Search from "../Navbar/Search";
import UserAccountSettings from "../Navbar/Buttons/UserAccountSettingsContainer";
import TabsHome from "./TabsHome";
import GridContainer from "../GridContainer";
import MenuRight from "../Navbar/MenuRight";
import WelcomePage from "./WelcomePageContainer";

type Props = {
  user: Object
};

function Home({ user }: Props) {
  if (user.firstAccess) {
    return <WelcomePage />;
  }

  return (
    <div>
      <Navbar>
        <div />
        <MenuRight>
          <Search />
          <UserAccountSettings />
        </MenuRight>
      </Navbar>
      <GridContainer>
        <TabsHome />
      </GridContainer>
    </div>
  );
}

export default Home;
