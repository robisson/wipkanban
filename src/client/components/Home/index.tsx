import * as React from "react";
import Navbar from "../Navbar";
import Search from "../Navbar/Search";
//import UserAccountSettings from "../Navbar/Buttons/UserAccountSettingsContainer";
import TabsHome from "./TabsHome";
import GridContainer from "../GridContainer";
import MenuRight from "../Navbar/MenuRight";
import WelcomePage from "./WelcomePageContainer";
import Preloader from "../PreloaderContainer";

interface IHomeProps {
  user: any
};

export default function Home({ user }: IHomeProps) {
  if (user.firstAccess) {
    return <WelcomePage />;
  }

  return (
    <div>
      <Preloader />
      <Navbar>
        <div />
        <MenuRight>
          <Search />

        </MenuRight>
      </Navbar>
      <GridContainer>
        <TabsHome />
      </GridContainer>
    </div>
  );
}
