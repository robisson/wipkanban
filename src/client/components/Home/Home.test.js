import React from "react";
import Home from "./index";
import Navbar from "../Navbar";
import { createShallow } from "@material-ui/core/test-utils";
import TabsHome from "./TabsHome";
import MenuRight from "../Navbar/MenuRight";
import Search from "../Navbar/Search";
import UserAccountSettings from "../Navbar/Buttons/UserAccountSettings";
import toJson from "enzyme-to-json";
import "../../setupTest";

describe("Home component", () => {
  let shallow;
  let user;

  beforeEach(() => {
    shallow = createShallow();
    user = {
      firstAccess: false
    };
  });

  it("should render 1 div containing Navbar", () => {
    let wrapper = shallow(<Home user={user} />);
    expect(wrapper.type()).toEqual("div");
    expect(wrapper.find(Navbar).length).toEqual(1);
    expect(wrapper.find(Navbar).length).toEqual(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a RightMenu with two icons component", () => {
    let wrapper = shallow(<Home user={user} />);

    expect(wrapper.find(MenuRight).type()).toEqual(MenuRight);
    expect(
      wrapper
        .find(MenuRight)
        .childAt(0)
        .type()
    ).toEqual(Search);
    expect(
      wrapper
        .find(MenuRight)
        .childAt(1)
        .type()
    ).toEqual(UserAccountSettings);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a TabHome component", () => {
    let wrapper = shallow(<Home user={user} />);

    expect(wrapper.find(TabsHome).type()).toEqual(TabsHome);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a WelcomePage component when user firstAccess is true", () => {
    user.firstAccess = true;
    let wrapper = shallow(<Home user={user} />);

    expect(wrapper.name()).toEqual("Connect(WithStyles(WelcomePage))");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
