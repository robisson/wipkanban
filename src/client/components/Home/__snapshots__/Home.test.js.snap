// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component should render 1 div containing Navbar 1`] = `
<div>
  <Connect(WithStyles(Preloader)) />
  <WithStyles(Navbar)>
    <div />
    <WithStyles(MenuRight)>
      <Search />
      <Connect(WithStyles(UserAccountSettings)) />
    </WithStyles(MenuRight)>
  </WithStyles(Navbar)>
  <WithStyles(GridContainer)>
    <TabsHome />
  </WithStyles(GridContainer)>
</div>
`;

exports[`<Home /> component should render a RightMenu with two icons component 1`] = `
<div>
  <Connect(WithStyles(Preloader)) />
  <WithStyles(Navbar)>
    <div />
    <WithStyles(MenuRight)>
      <Search />
      <Connect(WithStyles(UserAccountSettings)) />
    </WithStyles(MenuRight)>
  </WithStyles(Navbar)>
  <WithStyles(GridContainer)>
    <TabsHome />
  </WithStyles(GridContainer)>
</div>
`;

exports[`<Home /> component should render a TabHome component 1`] = `
<div>
  <Connect(WithStyles(Preloader)) />
  <WithStyles(Navbar)>
    <div />
    <WithStyles(MenuRight)>
      <Search />
      <Connect(WithStyles(UserAccountSettings)) />
    </WithStyles(MenuRight)>
  </WithStyles(Navbar)>
  <WithStyles(GridContainer)>
    <TabsHome />
  </WithStyles(GridContainer)>
</div>
`;

exports[`<Home /> component should render a WelcomePage component when user firstAccess is true 1`] = `<Connect(WithStyles(WelcomePage)) />`;
